pkg/querier/queryrange/results_cache.go

```diff
@@ -133,12 +139,41 @@ func (s resultsCache) Do(ctx context.Context, r Request) (Response, error) {
 	return response, err
 }
 
+// shouldCacheResponse says whether the response should be cached or not.
+func shouldCacheResponse(r Response) bool {
+	if promResp, ok := r.(*PrometheusResponse); ok {
+		shouldNotCache := false
```

Comment:
Just a minor nit, but I think it'll be clearer if we just do `shouldCache:= true` and setting it to false and just `return shouldCache`.